### PR TITLE
Hide nhs number question on check-answers page

### PR DESCRIPTION
**Pull request by leenagupte**
Trello: https://trello.com/c/wJEQ8R1f

# Expected changes
## User answers "No"
Screenshot 2020-03-24 at 17 28 58

## User answers "Yes"
Screenshot 2020-03-24 at 17 29 13

### CODE REVIEW

**Comment by leenagupte** [comment]
TO DO: Add tests

File: app/helpers/answers_helper.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    if question.eql?("contact_details")
+      concatenated_answer = []
+      concatenated_answer << "Phone number: #{answer['phone_number_calls']}" if answer["phone_number_calls"]
+      concatenated_answer << "Text: #{answer['phone_number_texts']}" if answer["phone_number_texts"]
+      concatenated_answer << "Email: #{answer['email']}" if answer["email"]
+      concatenated_answer.join("<br>")
+    elsif question.eql?("support_address")
+      answer.values.compact.join(",<br>")
+    elsif question.eql?("date_of_birth")
+      Time.zone.local(answer["year"], answer["month"], answer["day"]).strftime("%d/%m/%Y")
```
Comment (injms) [inline app/helpers/answers_helper.rb:36]: Ooh, this is nice!

File: spec/helpers/answers_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,132 @@
+require "spec_helper"
+
+RSpec.describe AnswersHelper, type: :helper do
+  describe "#answer_items" do
+    it "adds a query string to the link for each item" do
+      helper.answer_items.each do |item|
+        expect(item[:edit][:href]).to include("?change-answer")
```
Comment (injms) [inline spec/helpers/answers_helper_spec.rb:7]: Thanks for adding a test for this - I should've done that when I made the change earlier.

File: spec/helpers/answers_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    end
+
+    context "contact_details" do
+      let(:question) { "contact_details" }
+
+      it "concatenates contact_details with a line break" do
+        answer = {
+          "phone_number_calls" => "012101234567",
+          "phone_number_texts" => "0777001234567",
+          "email" => "me@me.com",
```
Comment (injms) [inline spec/helpers/answers_helper_spec.rb]: Nitpick - can we change this to 'example.com', as the `me.com` address exists and is used.

```suggestion
          "email" => "me@example.com",
```

File: spec/helpers/answers_helper_spec.rb [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+      it "returns nothing if the contact details are empty" do
+        answer = {}
+
+        expect(helper.concat_answer(answer, question)).to be_empty
+      end
+
+      it "only concatenates the fields that have a value" do
+        answer = {
+          "email" => "me@me.com",
```
Comment (injms) [inline spec/helpers/answers_helper_spec.rb]:
```suggestion
          "email" => "me@example.com",
```

File: spec/helpers/answers_helper_spec.rb (hunk 3 above)
Comment (leenagupte) [reply spec/helpers/answers_helper_spec.rb]: Sure! Added.

File: spec/helpers/answers_helper_spec.rb (hunk 2 above)
Comment (injms) [reply spec/helpers/answers_helper_spec.rb:7]: Actually, this should read thanks for adding all the tests that I should've done!

**Comment by leenagupte** [comment]
Thanks for the comments @injms. I've added your suggestions and fixed up.

**Review by injms — APPROVED** [review]
Super nice work 👍

**Pull request merged**